calculator/src/Components/Calculator/Calculator.js

```diff
@@ -0,0 +1,308 @@
+            } else {
+                this.setState({operands: [...newOperands, currentOperand + "."]});
+            }
+        }
+    } // End of addDecimal() function
+
+    addToScreen = (input) => {
+        let {operands, res} = this.state; // Grabbing all operations and operands from state
+        let lastIdx = operands.length - 1; // Grabbing the last index of the operands
+        let currentOperand = operands[lastIdx]; // Grabbing the current operand (last in the arr)
+        let newOperands = operands.slice(0, lastIdx); // Grabbing the operands arr w/o the current operand
+
+        if(this.isOperator(input)) {
+            // If the input is an operator, and the last input isn't an operator then the operator is set
+            if(!operatorLast) this.setOperator(input);
+
+        } else {
+            if(this.isDecimal(input)) {
+                this.addDecimal(input);
+            } else if(operands[0] === res.toString()) {
+                this.setState(({operands: [input], res: 0}));
+            }else {
+                if(currentOperand === "0") {
+                    this.setState({operands: [...newOperands, input]});
+                } else {
+                    this.setState({operands: [...newOperands, currentOperand + input]});
+                }
+                operatorLast = false;
+            }
+        }
+    } // End of addToScreen() function
+
+    isNegative = (operand) => operand[0] === "-";
+
+    changeValue = () => {
+        let {operands}  = this.state;
+        let lastIdx = operands.length - 1; // Grabbing the last index of the operands
+        let currentOperand = operands[lastIdx]; // Grabbing the current operand (last in the arr)
+        let newOperands = operands.slice(0, lastIdx); // Grabbing the operands arr w/o the current operand
+        
+        if(this.isNegative(currentOperand)) {
+            this.setState({operands: [...newOperands, currentOperand.slice(1)]})
+        } else {
+            if(currentOperand === "0") {
+                this.setState({operands: [...newOperands, "-"]})
+            } else {
+                this.setState({operands: [...newOperands, "-" + currentOperand]})
+            }
+        }
+    } // End of changeValue() function
+
+    changePercent = () => {
+        let {operands}  = this.state;
+        let lastIdx = operands.length - 1; // Grabbing the last index of the operands
+        let currentOperand = operands[lastIdx]; // Grabbing the current operand (last in the arr)
+        let newOperands = operands.slice(0, lastIdx); // Grabbing the operands arr w/o the current operand
+
+        this.setState({operands: [...newOperands, (currentOperand / 100).toString()]});
+    } // End of changePercent() function
+
+    resetScreen = () => { 
+        this.setState({operations: [], operands: ["0"]});
+    } // End of resetScreen() function
+
+    findMathFunction = (operation) => this.mathFunctions[operation]();
```

Comment:
Its cool you incorporated this, this is called an IIFE, i can you didnt know.

Heres more about it https://flaviocopes.com/javascript-iife/